unstable/lens/struct.rkt

```diff
@@ -0,0 +1,46 @@
+#lang racket/base
+
+(provide define-struct-lenses struct/lens)
+
+(require syntax/parse/define
+         lens/base/main
+         alexis/util/struct
+         (submod alexis/util/struct get-struct-accessors)
+         (for-syntax racket/base
+                     syntax/parse
+                     racket/syntax
+                     racket/struct-info
+                     ))
+(module+ test
+  (require rackunit fancy-app))
+
+(define-syntax define-struct-lenses
+  (syntax-parser
+    [(define-struct-lenses s:id)
+     #:do [(define-values (ids1 ids2)
+             (get-struct-accessors (extract-struct-info (syntax-local-value #'s)) #'s))]
+     #:with [s-fld:id ...] ids2
+     #:with [[s-fld-set s-fld-lens] ...]
+     (for/list ([s-fld (in-list ids2)])
+       (list (format-id #'s "~a-set" s-fld #:source s-fld)
+             (format-id #'s "~a-lens" s-fld #:source s-fld)))
+     #'(begin
+         (define-struct-updaters s)
+         (define s-fld-lens (make-lens s-fld s-fld-set))
+         ...)]))
```

Comment:
Can some of this functionality be broken up into one or more helper macros to simplify the implementation?